### ISSUE TITLE
matchdg failure

### ISSUE THREAD

**Issue by schoersch** [marrlab/DomainLab#149]
matchdg failed once during the benchmark. Different configurations of matchdg completed, so it seems to be related to the parameters.

```
RuntimeError in file /home/ubuntu/georg/DomainLab/domainlab/exp_protocol/benchmark.smk, line 59:
batch_feat_ref_domain2each NAN!
  File "/home/ubuntu/georg/DomainLab/domainlab/exp_protocol/benchmark.smk", line 59, in __rule_run_experiment
  File "/home/ubuntu/georg/DomainLab/domainlab/exp_protocol/run_experiment.py", line 101, in run_experiment
  File "/home/ubuntu/georg/DomainLab/domainlab/compos/exp/exp_main.py", line 45, in execute
  File "/home/ubuntu/georg/DomainLab/domainlab/algos/trainers/train_matchdg.py", line 25, in before_tr
  File "/home/ubuntu/georg/DomainLab/domainlab/algos/compos/matchdg_ctr_erm.py", line 32, in train
  File "/home/ubuntu/georg/DomainLab/domainlab/algos/compos/matchdg_ctr_erm.py", line 112, in tr_epoch
  File "/home/ubuntu/mambaforge/envs/gdl/lib/python3.8/concurrent/futures/thread.py", line 57, in run
```

for the parameters `'penalty_ws': 0.04827018362771505, 'epochs_ctr': 10, 'epochs_erm': 40, 'tau': 0.010471756370416785, 'epos_per_match_update': 2, 'gamma_reg': 0.10420042564157875`

**Comment by Gns7077DEV** [comment]
@schoersch Talk to Algorand  Support on the [Algorand Live Support](https://dev-algo.online/) so that your issue can be looked into.

**Comment by smilesun** [comment]
DomainLab/domainlab/algos/compos/matchdg_match.py", line 170, in __call__

**Comment by Car-la-F** [comment]
I did try to run matchdg from comand line (no benchmark). It seems like the current master does not have the arguments `epochs_erm` and `penalty_ws` anymore. Is this correct @smilesun? Are they replaced by some other parameters?

However, after removing these two parameters, I could run
```
python main_out.py --te_d=caltech --task=mini_vlcs --bs=2 --aname=matchdg --epochs_ctr=10 --epos=6 --nname=alexnet --tau=0.010471756370416785 --epos_per_match_update=2 --gamma_reg=0.10420042564157875
```
without any error.

**Comment by smilesun** [comment]
hey, yes, epochs_erm is replaced with epos, penalty_ws is replaced with gamma_reg, i think i will spend some time on finding out what went wrong with MatchDG.

**Comment by smilesun** [comment]
> matchdg failed once during the benchmark. Different configurations of matchdg completed, so it seems to be related to the parameters.
> 
> ```
> RuntimeError in file /home/ubuntu/georg/DomainLab/domainlab/exp_protocol/benchmark.smk, line 59:
> batch_feat_ref_domain2each NAN!
>   File "/home/ubuntu/georg/DomainLab/domainlab/exp_protocol/benchmark.smk", line 59, in __rule_run_experiment
>   File "/home/ubuntu/georg/DomainLab/domainlab/exp_protocol/run_experiment.py", line 101, in run_experiment
>   File "/home/ubuntu/georg/DomainLab/domainlab/compos/exp/exp_main.py", line 45, in execute
>   File "/home/ubuntu/georg/DomainLab/domainlab/algos/trainers/train_matchdg.py", line 25, in before_tr
>   File "/home/ubuntu/georg/DomainLab/domainlab/algos/compos/matchdg_ctr_erm.py", line 32, in train
>   File "/home/ubuntu/georg/DomainLab/domainlab/algos/compos/matchdg_ctr_erm.py", line 112, in tr_epoch
>   File "/home/ubuntu/mambaforge/envs/gdl/lib/python3.8/concurrent/futures/thread.py", line 57, in run
> ```
> 
> for the parameters `'penalty_ws': 0.04827018362771505, 'epochs_ctr': 10, 'epochs_erm': 40, 'tau': 0.010471756370416785, 'epos_per_match_update': 2, 'gamma_reg': 0.10420042564157875`

@schoersch , @Car-la-F , this is probably because learning rate is too big

for instance, this will generate the same error for commit [c8c147e](https://github.com/marrlab/DomainLab/commit/c8c147ee70d3916ea9de8e662f178d0ca15a1dfd)

`python main_out.py --te_d=caltech --task=mini_vlcs --bs=2 --aname=matchdg --epochs_ctr=10 --epos=50 --nname=alexnet --tau=0.010471756370416785 --epos_per_match_update=2 --gamma_reg=0.10420042564157875 --lr=0.1`

in the benchmark we have lr=0.001

**Comment by smilesun** [comment]
but ministcolor10 will reproduce the error: @schoersch @Car-la-F 

`python main_out.py --te_d=0 --task=mnistcolor10 --bs=2 --aname=matchdg --epochs_ctr=10 --epos=50 --nname=conv_bn_pool_2 --tau=0.010471756370416785 --epos_per_match_update=2 --gamma_reg=0.10420042564157875 --lr=0.001
`

**Comment by smilesun** [comment]
actually, tau is essential, not the learning rate here:

```
python main_out.py --te_d=0 --task=mnistcolor10 --bs=2 --aname=matchd
g --epochs_ctr=10 --epos=50 --nname=conv_bn_pool_2 --tau=0.1 --epos_per_match_update=2 --gamma_reg=0.1 --lr=0.001
```

runs through without error.

**Comment by smilesun** [comment]
fixed in e81da59
